src/hooks/useSelector.js

```diff
@@ -68,8 +71,8 @@ function useSelectorWithStoreAndSubscription(
         // changed
         latestSubscriptionCallbackError.current = err
       }
-
-      forceRender({})
+      latestReduxState.current = undefined
+      forceRender(newReduxState)
```

Comment:
I'm not sure this achieves what you want. The state that you set here is only used when the selector changes on the next render or an error was thrown above. In all other cases it is still `latestSelectedState.current` that gets used for rendering. And `latestSelectedState.current` is not safe in CM, since it may have been updated while a render that depends on an old value is suspended.

Reply:
When I implemented this originally, I experimented with putting `latestSelectedState` in a `useState` instead of using a ref, but I didn't go through with that approach for a reason I can't remember sadly.

Looking at it now with fresh eyes I think the solution may be to store both, the latest full store state as well as the latest selected state in a `useState` and update each depending on whether calling the selector succeeded or not. This way there should never be a reason to access a state ref during rendering.

Reply:
All of that said, I believe that the CM issues are more fundamental. Basically, any code that uses refs is very likely to be susceptible to tearing or other issues. In this hook for example, imagine someone changes the selector. However, the render that changes the selector is suspended. During suspension a store update is processed. Since `checkForUpdates` uses the selector from the latest committed render, it will use a stale selector for processing the update, which may return in missed updates.

Reply:
The core problems remain the same, i.e. that store updates and rendering happen in two different "logical threads" so to speak that are prone to unfortunate interleaving. I believe the only proper way to fix this is to use the context to propagate the store value, which will hopefully become feasible once context selectors are implemented.

Reply:
Let me put a link how I did approach to the CM issue (at least one of issues). https://github.com/dai-shi/reactive-react-redux/issues/29

Reply:
Hi, that assignment is NOT on the render phase, is on the subscription Callback.
We have 2 case when store changes

the change produce a selector change -> we enqueque a component render with useState.setState(newReduxStore)
the change DOES not produce a selector change -> we "save" the new reduxState, because we have to always remember the last "reduxState event callback change" if there is an update from others reason like prop changes.

Reply:
> The core problems remain the same, i.e. that store updates and rendering happen in two different "logical threads" so to speak that are prone to unfortunate interleaving. I believe the only proper way to fix this is to use the context to propagate the store value, which will hopefully become feasible once context selectors are implemented.

that will come back to v6.x performance issue. hum.